deepspeed/module_inject/containers/gptj.py

```diff
@@ -77,21 +77,33 @@ def __init__(self, client_module, inference=True):
         self.client_module = client_module
         try:
             import transformers
-            HFGPTJLayerPolicy._orig_layer_class = transformers.models.gptj.modeling_gptj.GPTJBlock
+            HFGPTJLayerPolicy._orig_layer_class = [
+                transformers.models.gptj.modeling_gptj.GPTJBlock,
+                transformers.models.codegen.modeling_codegen.CodeGenBlock
+            ]
         except:
             HFGPTJLayerPolicy._orig_layer_class = None
 
     def get_hidden_heads(self):
-        return self.client_module.attn.q_proj.weight.shape[1], \
+        return self.client_module.attn.out_proj.weight.shape[1], \
```

Comment:
Does this not break gpt-j? Gptj and codegen have different naming schemes for qkv. When I was briefly looking into adding codegen support I created a separate container file like this https://github.com/microsoft/DeepSpeed/compare/master...mosm/codegen

Reply:
but I am using out_proj here, and gptj has the same naming convention: https://github.com/huggingface/transformers/blob/main/src/transformers/models/gptj/modeling_gptj.py#L115